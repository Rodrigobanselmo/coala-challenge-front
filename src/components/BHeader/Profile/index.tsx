// import { Avatar, Box, Flex, Text, useBreakpointValue } from '@chakra-ui/react';

import Avatar from "@mui/material/Avatar";
import Box from "@mui/material/Box";
import Text from "@mui/material/Typography";

import { IProfileProps } from "./types";
import { useAuth } from "../../../core/contexts/AuthContext";

export function Profile({}: IProfileProps): JSX.Element {
  const { user } = useAuth();

  const name = user?.name || "";
  const email = user?.email || "";

  return (
    <Box style={{ alignItems: "center", display: "flex", cursor: "pointer" }}>
      {name && (
        <Box mr={8} textAlign="right">
          <Text color={"text.main"} sx={{ fontSize: "0.99rem" }}>
            {name}
          </Text>
          <Text
            color={"text.main"}
            mt={-1}
            sx={{ opacity: 0.5, fontSize: "0.81rem" }}
          >
            {email}
          </Text>
        </Box>
      )}
      <Avatar
        src={user?.photoUrl || "/icons/profile.svg"}
        alt={name}
        sx={{
          backgroundColor: "gray.700",
          width: ["32px", "48px"],
          height: ["32px", "48px"],
          "& .MuiAvatar-img": {
            backgroundColor: "background.default",
            objectFit: "cover",
          },
        }}
      >
        {name.split(" ")[0][0]}
        {name.split(" ")[1]?.[0] || ""}
      </Avatar>
    </Box>
  );
}
