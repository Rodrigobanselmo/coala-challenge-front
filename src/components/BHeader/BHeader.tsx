import MenuIcon from "@mui/icons-material/Menu";
import IconButton from "@mui/material/IconButton";
import Stack from "@mui/material/Stack";

import { Profile } from "./Profile";
import { useSidebarDrawer } from "@/core/contexts/SidebarContext";
import { Typography } from "@mui/material";
import { useRouter } from "next/router";
import { drawerMap } from "@/constants/drawer.constants";
import { Session } from "./Session";

export function BHeader(): JSX.Element {
  const { open, isTablet } = useSidebarDrawer();

  return (
    <Stack
      direction="row"
      component="header"
      mx={[2, 4, 6]}
      px={[4, 6, 8]}
      sx={{
        borderBottom: "1px solid",
        borderColor: "gray.200",
        alignItems: "center",
        height: ["3rem", "4rem", "5rem"],
      }}
    >
      <Session />

      {isTablet && (
        <IconButton
          aria-label="Open navigation"
          onClick={open}
          sx={{
            ml: "-3px",
          }}
        >
          <MenuIcon sx={{ fontSize: "1.6rem" }} />
        </IconButton>
      )}

      <Stack
        direction="row"
        sx={{
          alignItems: "center",
        }}
        ml="auto"
      >
        <Profile />
      </Stack>
    </Stack>
  );
}
