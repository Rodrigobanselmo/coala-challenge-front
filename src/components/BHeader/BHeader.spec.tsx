import React from "react";
import { render } from "@testing-library/react";
import { BHeader } from "./BHeader";

jest.mock("../../core/contexts/AuthContext", () => ({
  useAuth: () => ({
    user: {
      id: "1",
      email: "qwe@qwe.com",
    },
  }),
}));

jest.mock("../../core/contexts/SidebarContext", () => ({
  useSidebarDrawer: () => ({
    open: () => {},
    isTablet: true,
  }),
}));

describe("BHeader", () => {
  it("matches the snapshot", () => {
    const { container } = render(<BHeader />);
    expect(container).toMatchSnapshot();
  });
});
