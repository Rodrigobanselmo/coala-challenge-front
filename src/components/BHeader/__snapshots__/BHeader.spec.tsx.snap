// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BHeader matches the snapshot 1`] = `
<div>
  <header
    class="MuiStack-root css-14e3gqt-MuiStack-root"
  >
    <div
      class="MuiBox-root css-k008qs"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-tes3bw-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Sair
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pezyni-MuiTypography-root"
      >
        Fializar Sess√£o
      </p>
    </div>
    <button
      aria-label="Open navigation"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-wxlioq-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tp5iyz-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiStack-root css-wwyfu-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
        style="align-items: center; display: flex; cursor: pointer;"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-8v8enj-MuiAvatar-root"
          data-testid="BAvatar"
        >
          <img
            alt=""
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="/icons/profile.svg"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
